app/src/lib/stores/app-store.ts

```diff
@@ -762,11 +762,13 @@ export class AppStore extends TypedBaseStore<IAppState> {
         return
       }
 
+      const newState: IDisplayHistory = {
+        kind: HistoryTabMode.History,
+      }
+
       this.repositoryStateCache.updateCompareState(repository, () => ({
         tip: currentSha,
-        formState: {
-          kind: HistoryTabMode.History,
-        },
+        formState: newState,
```

Comment:
Why these changes? What kind of errors were thrown?

Reply:
This change is because the type inference couldn't verify this satisfies the criteria of `formState`:

> Type '{ kind: HistoryTabMode; }' is not assignable to type 'IDisplayHistory | ICompareBranch'.

If I break it out it becomes clearer where the problem lies:

<img width="262" src="https://user-images.githubusercontent.com/359239/50493095-b3cf8200-09f1-11e9-9c29-5667d4cf5a50.png">

The next time we're in making changes to the compare tab we can revisit these abstractions, but for now I just want to appease the Type System Gods™